app/navigation/nonsipp/SharesDisposalNavigator.scala

```diff
@@ -148,13 +149,190 @@ object SharesDisposalNavigator extends JourneyNavigator {
         .onPageLoad(srn, shareIndex, disposalIndex, NormalMode)
 
     case HowManySharesPage(srn, shareIndex, disposalIndex) =>
+      controllers.nonsipp.sharesdisposal.routes.SharesDisposalCYAController
+        .onPageLoad(srn, shareIndex, disposalIndex, NormalMode)
+
+    case SharesDisposalCompletedPage(srn, shareIndex, disposalIndex) =>
       controllers.routes.UnauthorisedController.onPageLoad()
 
   }
 
   override def checkRoutes: UserAnswers => UserAnswers => PartialFunction[Page, Call] =
     _ =>
-      _ => {
-        PartialFunction.empty
+      userAnswers => {
+
+        case page @ HowWereSharesDisposedPage(srn, shareIndex, disposalIndex, _) => //c
+          userAnswers.get(page) match {
+            case None =>
+              controllers.routes.JourneyRecoveryController.onPageLoad()
+            case Some(HowSharesDisposed.Sold) =>
+              controllers.nonsipp.sharesdisposal.routes.WhenWereSharesSoldController //d
+                .onPageLoad(srn, shareIndex, disposalIndex, CheckMode)
+            case Some(HowSharesDisposed.Redeemed) =>
+              controllers.nonsipp.sharesdisposal.routes.WhenWereSharesRedeemedController //l
+                .onPageLoad(srn, shareIndex, disposalIndex, CheckMode)
+            case Some(HowSharesDisposed.Transferred) | Some(HowSharesDisposed.Other(_)) =>
+              controllers.nonsipp.sharesdisposal.routes.SharesDisposalCYAController
+                .onPageLoad(srn, shareIndex, disposalIndex, NormalMode)
```

Comment:
Do we need to open the CYA page in check mode or normal mode ? 

This comments are for all CYA navigation under checkRoutes

Reply:
Good spot - when working on / looking at CYA pages and navigation in the past, I've noticed that the usage of the Check / Change variants is inconsistent, and this has been raised in one or more bug tickets, but I don't think it's been resolved or made consistent across the whole service yet. Having the "Check" page as the destination for all CheckMode navigation is what was implemented in a previous CYA journey I looked at, so I continued that pattern here, but I agree that it's not ideal and should be fixed across the service.

Incidentally, if the PointOfEntry code is kept, we might be able use that to fix these Check / Change inconsistencies across the whole service.